src/Executables/ExtendConnectivity/ExtendConnectivity.cpp

```diff
@@ -0,0 +1,85 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <array>
+#include <boost/program_options.hpp>
+#include <cstddef>
+#include <string>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "IO/H5/AccessType.hpp"
+#include "IO/H5/File.hpp"
+#include "IO/H5/Helpers.hpp"
+#include "IO/H5/VolumeData.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Parallel/Printf.hpp"
+
+// Charm looks for this function but since we build without a main function or
+// main module we just have it be empty
+extern "C" void CkRegisterMainModule(void) {}
+
+void extend_connectivity(const std::string& file_name,
+                         const std::string& subfile_name) {
+  h5::H5File<h5::AccessType::ReadWrite> data_file(file_name, true);
+  auto& volume_file = data_file.get<h5::VolumeData>("/" + subfile_name);
+  const std::vector<size_t>& observation_ids =
+      volume_file.list_observation_ids();
+  const size_t dim = volume_file.get_extents(observation_ids[0])[0].size();
+
+  switch (dim) {
+    case 1:
+      volume_file.extend_connectivity_data<1>(observation_ids);
+      break;
+    case 2:
+      volume_file.extend_connectivity_data<2>(observation_ids);
+      break;
+    case 3:
+      volume_file.extend_connectivity_data<3>(observation_ids);
+      break;
+    default:
+      ERROR("Invalid dimensionality");
```

Comment:
Maybe stream out the dim?

Reply:
Would the syntax be something like

```
ERROR("Invalid dimensionality. The given dimensionality was: " << dim);
```

where `dim` is the variable storing the value parsed by the function to determine the dimensionality of the data.